misk-jobqueue/src/main/kotlin/misk/jobqueue/JobConsumer.kt

```diff
@@ -3,15 +3,21 @@ package misk.jobqueue
 /**  A [JobConsumer] allows applications to subscribe to receive incoming jobs */
 interface JobConsumer {
   /**
-   * Registers a handler to receive messages. Once registered, the consumer will immediately
-   * begin receiving messages from the underlying job queue and dispatch them to the provided
-   * handler. A service may only have one subscription outstanding per queue
+   * Associates a message handler to a prioritized list of queues. Once registered, the consumer
+   * will immediately begin receiving messages from the underlying job queues and dispatch them to
+   * the provided handler.
+   *
+   * If more than one queue name is provided, these will be treated as a list of prioritized queues
```

Comment:
we should be explicit about the chance of starvation for the lower priority queues. for example in the case of the higher priority queues being over saturated or if the higher priority queues are failing to process.

Reply:
Yeah, good call. I'll update the docs. Thanks!

Reply:
It seems like that as long as there's a constant trickle of messages going into the higher priority queues, the lower priorities queues will never processed?

Reply:
this should only happen if we're processing messages at the maximum rate (so it's correct to starve the lower-priority queue in this case), though there is an issue where we currently wait for all workers to finish before pulling the next batch. i have another PR to address that.

Reply:
Each Receive request can contain _up to_ 10 messages. If you get 1 message in each Receive request (a trickle), you'll be processing at 1/10 rate and starve the low-priority queues.